fpga/Makefile

```diff
@@ -120,6 +137,84 @@ debug-bitstream: $(build_dir)/obj/post_synth.dcp
 			$(build_dir)/debug_obj \
 			$(fpga_common_script_dir)
 
+#########################################################################################
+# sim rules
+#########################################################################################
+
+# hack until able to restructure fpga-shells .tcl scripts
+project-creation-script := $(fpga_common_script_dir)/create_project.tcl
+$(project-creation-script): $(fpga_common_script_dir)/vivado.tcl
+	cd $(fpga_common_script_dir); sed -n '1,10p' vivado.tcl > create_project.tcl
+
+# generate the vivado project and add all sources
+vivado-project := $(build_dir)/$(MODEL).xpr
+$(vivado-project): $(synth_list_f) $(project-creation-script)
+	cd $(build_dir); vivado \
+		-nojournal -mode batch \
+		-source  $(project-creation-script) \
+		-tclargs \
+			-top-module "$(MODEL)" \
+			-F "$(synth_list_f)" \
+			-ip-vivado-tcls "$(shell find '$(build_dir)' -name '*.vivado.tcl')" \
+			-board "$(BOARD)"
+	# clean up the hacked project creation script
+	rm $(project-creation-script)
+
+# generate vcs collateral
+vivado-vcs-collateral := $(build_dir)/$(MODEL).cache/compile_simlib/synopsys_sim.setup
+$(vivado-vcs-collateral): $(sim_dir)/scripts/generate_vcs_collateral.tcl $(vivado-project)
+	cd $(build_dir); vivado $(vivado-project) \
+		-nojournal -mode batch \
+		-source $(sim_dir)/scripts/generate_vcs_collateral.tcl \
+		-tclargs \
+			$(FPGA_FAMILY) \
```

Comment:
I'm surprised that you don't need to pass the board itself (like Arty A7)...

Reply:
The Vivado tcl command in `generate_vcs_collateral.tcl` that takes the `FPGA_FAMILY` is `compile_simlib`, which compiles behavioral models for primitives (such as IOBUFs) which fpga-shells blackboxes. These are specific to an FPGA family (e.g. Ultrascale, Kintex), and not peripherals of the dev board on which the FPGA lives.